src/components/Form.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import React, { useState } from 'react';
+
+type FormProps = {
+  onSubmit: (value: number) => void;
+};
+
+export default function Form({ onSubmit }: FormProps) {
+  const [inputValue, setInputValue] = useState('');
+  const [error, setError] = useState('');
+  const isDisabled = inputValue === '';
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    const sanitizedValue = value.replace(/\D/g, '');
+
+    if (sanitizedValue === '') {
+      setInputValue('');
+      setError('');
+      return;
+    }
+
+    const parsed = Number(sanitizedValue);
+    if (parsed < 0) {
+      setError('0以上の整数を入力してください');
+      return;
+    }
+
+    setError('');
+    setInputValue(sanitizedValue);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    
+    if (inputValue === '') {
+      setError('数値を入力してください');
+      return;
+    }
+
+    const parsed = Number(inputValue);
+
+    onSubmit(parsed);
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="flex flex-col gap-4 w-80">
+      <input
+        type="text"
```

Comment:
[imo]
最初にお伝えできずすみませんmm
アクセシビリティの観点から、input タグには label タグをセットで記述した方がいいです。

```suggestion
      <label for="follower">フォロワー数</label>
      <input
        id="follower"
        type="text"
```

参考
- [アクセシビリティ第一歩、labelが超重要な話。~ placeholderじゃダメなの？~](https://zenn.dev/nako_110/articles/ce7e9a453dfd07)

Reply:
`<label htmlFor="follower" className="sr-only">フォロワー数</label>`
UI上の観点から、表示はしない方向で設定